### ISSUE TITLE
Git LFS support for build agent

### ISSUE THREAD

**Issue by Tasteful** [Microsoft/vsts-agent#554]
## **Agent version and platform**
Default agent that is included in TFS 2015.3, Agent.Version 1.95.3

OS of the machine running the agent? Windows 2012r2

## **VSTS type and version**
On-Prem TFS 2015.3

## **What's not working?** 
I trying to get a build to work and in the git repository I'm using Git LFS files that not is fetched after checkout. TFS 2015.3 have builtin Git LFS support but the build agent is not restoring the content correctly. Is there any extra step I need to do to get this to work?

**Comment by bryanmacfarlane** [comment]
No, it's not there yet.  Step 1 is move to new combined agent which uses git.exe (this project).  After shipping this, next steps are enhancements like shallow clone, LFS, etc...

**Comment by Tasteful** [comment]
@bryanmacfarlane Is this agent today compatible with TFS2015.3 or do I need to run TFS.vNext?

**Comment by Tasteful** [comment]
I have tested the VSTS agent with TFS2015.3 and it's working as it should. Not sure if all different tasks are working.

When the `git-lfs.exe` was in the path on the server it was automatic trying to download the LFS content with authentication failure. The git credential manager was starting in background and ask for username/password so need to globaly turn that off.

Using the OAuth token as password was not working for on my local installation (found that as suggestion in some other issue). 

I created a workaround

* remove the global git-lfs filters 
* created a script that was fetch/checkout the LFS files with a username/password that I set as build variable.

Created an issue at the git-lfs repo if they can attach the same extra-headers that VSTS is using to pass the OAuth-token to the server, issue is found here https://github.com/github/git-lfs/issues/1500.
